### PR TITLE
Add a new Jenkins job to run the kyua testsuite

### PR DESCRIPTION
**Pull request by arichardson**

### CODE REVIEW

**Comment by bsdjhb** [comment]
Hmm, how often does this try to build, and have you timed a run to see how long it takes?  If I read it correctly this throttles at once per hour, but at least on my (whimpy) desktop it took multiple hours to complete a single run under QEMU.  I feel like we might want to run these tests once a day at most?

**Comment by arichardson** [comment]
> Hmm, how often does this try to build, and have you timed a run to see how long it takes? If I read it correctly this throttles at once per hour, but at least on my (whimpy) desktop it took multiple hours to complete a single run under QEMU. I feel like we might want to run these tests once a day at most?

I was planning to do it once per day and only for master+dev+maybe some other branches. But I lost that change in the github webui.
The old jenkins job split the testsuite into a few different parallel jobs which we can also do here. However, that just uses a different root kyuafile. Having kyua support sharding natively would be much better.

**Comment by jrtc27** [comment]
Probably more like once or twice a week? Most changes are unlikely to affect the test suite.

**Comment by bsdjhb** [comment]
master + dev makes sense.  The purecap kernel branch (until it is merged to dev) might also be one to check.  I would be happy with once a week.

**Comment by jrtc27** [comment]
I think we want the same set of branches that CheriBSD-pipeline watches, which is:
- master
- dev
- cheri-purecap-kernel
- upstream-llvm-merge
- caprevoke (added by me a couple of days ago)
Any kyua differences in those are of interest. The upstream-llvm-merge branch is normally short-lived but I think it makes sense to have that gate a new LLVM version (or at least _inform_ whether or not we merge). If no source changes to CheriBSD are needed for a given LLVM merge we can still create the branch and just not make any changes.

File: Jenkinsfile.testsuite [hunk 1]
```diff
@@ -0,0 +1,55 @@
+@Library('ctsrd-jenkins-scripts') _
+
+// Set job properties:
+def jobProperties = [rateLimitBuilds([count: 1, durationName: 'hour', userBoost: true]),
+                     [$class: 'GithubProjectProperty', displayName: '', projectUrlStr: 'https://github.com/CTSRD-CHERI/cheribsd/'],
+                     copyArtifactPermission('*'), // Downstream jobs need the kernels/disk images
```
Comment (jrtc27) [inline Jenkinsfile.testsuite]: Not for this one they don't

**Comment by arichardson** [comment]
TODO: should probably build stuff instead of copying once the job actually works. Copying is there to speed up testing.

File: Jenkinsfile.testsuite (hunk 1 above)
Comment (jrtc27) [reply Jenkinsfile.testsuite]: ... and fixed whilst I was writing that :D

Comment (arichardson) [reply Jenkinsfile.testsuite]: copy-pasted and was fixed for run `#4`

Comment (arichardson) [reply Jenkinsfile.testsuite]: Jenkins testing is kinda awkward so I just keep using replay+jenkins pipeline unit stuff.

File: Jenkinsfile [hunk 2]
```diff
@@ -2,22 +2,35 @@
...
+    public static boolean COPY_INSTEAD_OF_BUILD_FIXME_REMOVE_THIS = true;
 }
 
 if (env.CHANGE_ID && !shouldBuildPullRequest()) {
     echo "Not building this pull request."
     return
 }
 
+echo("JOB_NAME='${env.JOB_NAME}', JOB_BASE_NAME='${env.JOB_BASE_NAME}'")
+def rateLimit = rateLimitBuilds([count: 1, durationName: 'day', userBoost: true])
```
Comment (arichardson) [inline Jenkinsfile:15]: It appears that `rateLimitBuilds()` is broken with the current jenkins installation: `WARNING: Unknown parameter(s) found for class type 'jenkins.branch.RateLimitBranchProperty$JobPropertyImpl': count,durationName,userBoost`
This is also happening on the normal job so the ratelimit is not being applied :( 
Guess I'll have to open a Jenkins bug report.

**Pull request merged**